### PR TITLE
Get claims by target ID

### PR DESCRIPTION
**Pull request by bogdan-zaharia**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/DOC-1100

### *What changes have we introduced*

- Added gateway function to retrieve all claims by target ID
- Added use case to retrieve all claims by target ID
- Added new endpoint to GET all claims by target ID
- Updated the `ClaimRequest` validator to only accept `Name` between 1 - 300 characters
- Added unit and integration tests for all of the above

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

This new endpoint needs to be locked down so it is only accessible to the MMH team.

### CODE REVIEW

**Comment by andrewclowes** [comment]
We might benefit from adding an index on the `targetId` column.

File: DocumentsApi/V1/UseCase/GetClaimsByTargetIdUseCase.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            _documentsGateway = documentsGateway;
+        }
+
+        public Dictionary<string, List<ClaimResponse>> Execute(Guid targetId)
+        {
+            var claims = _documentsGateway.FindClaimsByTargetId(targetId);
+
+            if (claims == null)
+            {
+                throw new NotFoundException($"No claims have been found for target ID: {targetId}");
```
Comment (andrewclowes) [inline DocumentsApi/V1/UseCase/GetClaimsByTargetIdUseCase.cs:26]: Is it appropriate to return a 404 from the endpoint if no search results are found? The `GET /api/v1/claims` resource does exist and it is only the query that returned no results. Perhaps we should be returning a 200 with an empty collection in the response.

File: DocumentsApi/V1/Gateways/DocumentsGateway.cs [hunk 2]
```diff
@@ -58,6 +60,19 @@ public Claim FindClaim(Guid id)
             return entity.ToDomain();
         }
 
+        public List<Claim> FindClaimsByTargetId(Guid targetId)
+        {
+            var entities = _databaseContext.Claims.Where(claimEntity => claimEntity.TargetId == targetId).Include(claimEntity => claimEntity.Document);
+            if (!entities.Any()) return null;
```
Comment (andrewclowes) [inline DocumentsApi/V1/Gateways/DocumentsGateway.cs:66]: Do we need this line? Could we just return the empty `Claim` collection and, in turn, simplify the use case?

Comment (bogdan-zaharia) [reply DocumentsApi/V1/Gateways/DocumentsGateway.cs]: Addressed

**Review by andrewclowes — APPROVED** [review]

**Pull request merged**